app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def github
+    user = User.from_omniauth(request.env['omniauth.auth'])
+    redirect_to root_path, alert: "you're not registered on our website" and return if user.nil?
```

Comment:
Comme la ligne commence à être longue j’aurais plutôt tendance à l’écrire

```suggestion
    if user.nil?
      return redirect_to(root_path, alert: "You’re not registered on our website.")
    end
```

Reply:
ah j'ai tendance a laisser les guards en une seule ligne quoi qu'il en soit, mais effectivement c'est difficilement lisible..
Aussi j ne savais pas qu'on pouvait 'return redirect_to', j'utilise toujours 'and return'!